### PR TITLE
fix(allocation-policies): delete policies to admin re-do

### PR DESCRIPTION
**Pull request by MeredithAnya**
The UI was supposed to get added in https://github.com/getsentry/snuba/pull/6248, not sure what happened here. 

Also had to fix the endpoint too - I think I did a bad merge conflict resolution or something

Screenshot 2024-09-26 at 12 05 44 PM

### CODE REVIEW

File: snuba/admin/views.py
Comment (kylemumma) [inline snuba/admin/views.py:1]: im confused what this deletion is for?

Comment (MeredithAnya) [reply snuba/admin/views.py:1]: this was a result of a bad merge conflict - it overwrites the logic that adds the `query_type`. it was supposed to be deleted before

**Comment by onkar** [comment]
Can you attach the screenshot of how this change looks like in the UI? Just want to make sure it does the right thing in like in https://github.com/getsentry/snuba/pull/6248.

**Comment by MeredithAnya** [comment]
> Can you attach the screenshot of how this change looks like in the UI? Just want to make sure it does the right thing like in #6248.

Deletes look like (there wont be a header if no delete allocation policies exist): 

Screenshot 2024-09-26 at 12 06 44 PM

**Review by kylemumma — APPROVED** [review]

**Pull request merged**